ang/civicase/Actions.js

```diff
@@ -1,386 +1,364 @@
 (function (angular, $, _) {
   var module = angular.module('civicase');
 
-  module.directive('civicaseActions', function (dialogService) {
+  module.directive('civicaseActions', function (dialogService, PrintMergeCaseAction) {
     return {
       restrict: 'A',
-      template:
-      '<li ng-class="{disabled: !isActionEnabled(action)}" ng-if="isActionAllowed(action)" ng-repeat="action in caseActions">' +
-      '  <a href ng-click="doAction(action)"><i class="fa {{action.icon}}"></i> {{ action.title }}</a>' +
-      '  <ul ng-if="isHasSubMenu(action)" class="dropdown-menu sub-menu">' +
-      '    <li ng-class="{disabled: !isActionEnabled(subMenu)}" ng-if="isActionAllowed(subMenu)" ng-repeat="subMenu in action.items">' +
-      '      <a href ng-click="doAction(subMenu)"><i class="fa {{subMenu.icon}}"></i> {{ subMenu.title }}</a>' +
-      '    </li>' +
-      '  </ul>' +
-      '</li>',
+      templateUrl: '~/civicase/Actions.html',
       scope: {
         cases: '=civicaseActions',
         refresh: '=refreshCallback',
         popupParams: '='
       },
-      link: function ($scope, element, attributes) {
-        var ts = CRM.ts('civicase');
-        var multi = $scope.multi = attributes.multiple;
-
-        $scope.isHasSubMenu = function(action) {
-          return (action.items && action.items.length);
-        };
-
-        $scope.isActionEnabled = function (action) {
-          return (!action.number || $scope.cases.length === +action.number);
-        };
-
-        $scope.isActionAllowed = function (action) {
-          var isLockCaseAction = _.startsWith(action.action, 'lockCases');
-          var isCaseLockAllowed = CRM.civicase.allowCaseLocks;
-
-          return (isLockCaseAction && isCaseLockAllowed) ||
-            (!isLockCaseAction && (!action.number || ((multi && action.number > 1) || (!multi && action.number === 1))));
-        };
-
-        // Perform bulk actions
-        $scope.doAction = function (action) {
-          if (!$scope.isActionEnabled(action)) {
-            return;
-          }
-
-          var result = $scope.$eval(action.action, {
-            editTags: function (item) {
-              var keys = ['tags'];
-              var model = {
-                tags: []
-              };
+      link: civicaseActionsLink
+    };
 
-              _.each(CRM.civicase.tagsets, function (tagset) {
-                model[tagset.id] = [];
-                keys.push(tagset.id);
-              });
+    /**
+     * Angular JS's link function for civicaseActions Directive
+     * @param {Object} $scope
+     * @param {Object} element
+     * @param {Object} attributes
+     */
+    function civicaseActionsLink ($scope, element, attributes) {
+      var ts = CRM.ts('civicase');
+      var multi = $scope.multi = attributes.multiple;
```

Comment:
Can't we just save `attribute.multiple` in `multi` var and save `$scope.multi`, as we are not using this variable in the template ( I guess)
```
var multi = attributes.multiple;
```

Reply:
Also, if we can have better naming for this var for improved readability?
`supportsMultipleValues` maybe ?

Reply:
Renamed to `isBulkMode`. Also removed the `$scope` variable as it was not being used.